rosys/actors/detector.py

```diff
@@ -0,0 +1,82 @@
+import socketio
+from ..world import BoxDetection, PointDetection
+from .actor import Actor
+from .. import event, task_logger
+from ..world import Frame
+
+
+class Detector(Actor):
+    interval: float = 0
+
+    def __init__(self):
+        super().__init__()
+        self.sio = socketio.AsyncClient()
+        self.is_connected = None
+
+        @self.sio.event
+        def connect():
+            self.is_connected = True
+            assert self.sio.transport() == 'websocket'
+
+        @self.sio.event
+        def connect_error(data):
+            self.is_connected = False
+
+        @self.sio.event
+        def disconnect():
+            self.is_connected = False
+
+    async def connect(self, reconnect_delay: float = 3):
+        try:
+            self.log.info('connecting to detector')
+            await self.sio.connect('ws://localhost:8004', socketio_path='/ws/socket.io')
+            self.log.info('connected successfully')
+            self.is_connected = True
+        except:
+            self.log.exception('connection failed; trying again')
+            self.is_connected = None
+            await self.sleep(reconnect_delay)
+
+    async def step(self):
+        if self.is_connected is None:
+            await self.connect()
+
+        if self.world.upload_queue:
+            task_logger.create_task(self.upload(self.world.upload_queue.pop(0)), name='upload_frame')
```

Comment:
There used to be a throttling using `minimal_minutes_between_uploads`. Is it not needed anymore?